build.sbt

```diff
@@ -851,3 +850,154 @@ commands ++= {
 }
 
 addCommandAlias("scalap",   "scalap/compile:runMain              scala.tools.scalap.Main -usejavacp")
+
+lazy val intellij = taskKey[Unit]("Update the library classpaths in the IntelliJ project files.")
+
+def moduleDeps(name: String) = Def.taskDyn { Def.task((name, (externalDependencyClasspath in Compile in LocalProject(name)).value.map(_.data))) }
```

Comment:
This should be doable without a dynamic task:

```
def moduleDeps(name: String) = (externalDependencyClasspath in Compile in LocalProject(name)).map(a => (name, a.map(_.data)))
```

You can also get rid of the dynamic references but our project names were chosen poorly so they clash with sbt imports (and there's no way to re-import the correct ones). We'd have to rename the projects or use an alias:

```
def moduleDeps(p: Project) = (externalDependencyClasspath in Compile in p).map(a => (p.id, a.map(_.data)))
def compilerP = compiler
def testP = test

...

    buildModule :: List(
      moduleDeps(actors).value,
      moduleDeps(compilerP).value,
      // moduleDeps("dist").value,                // No sources, therefore no module in IntelliJ
      moduleDeps(forkjoin).value,
      moduleDeps(interactive).value,
      moduleDeps(junit).value,
      moduleDeps(library).value,
      // moduleDeps("library-all").value,         // No sources
      moduleDeps(manual).value,
      moduleDeps(partestExtras).value,
      moduleDeps(partestJavaAgent).value,
      moduleDeps(reflect).value,
      moduleDeps(repl).value,
      moduleDeps(replJline).value,
      // moduleDeps("repl-jline-embedded").value, // No sources
      // moduleDeps("root").value,                // No sources
      // moduleDeps("scala-dist").value,          // No sources
      moduleDeps(scaladoc).value,
      moduleDeps(scalap).value,
      moduleDeps(testP).value)
```